crates/tx5-connection/src/webrtc/libdatachannel.rs

```diff
@@ -213,43 +213,27 @@ async fn task(
 
 async fn task_err(
     is_polite: bool,
-    config: Vec<u8>,
+    config: WebRtcConfig,
     send_buffer: usize,
     mut evt_send: CloseSend<WebrtcEvt>,
     cmd_send: CloseSend<Cmd>,
     mut cmd_recv: CloseRecv<Cmd>,
 ) -> Result<()> {
     evt_send.set_close_on_drop(true);
 
-    #[derive(Debug, Default, serde::Deserialize)]
-    #[serde(rename_all = "camelCase")]
-    struct U {
-        pub urls: Vec<String>,
-    }
-
-    #[derive(Debug, serde::Deserialize)]
-    #[serde(rename_all = "camelCase")]
-    struct C {
-        #[serde(default)]
-        pub ice_servers: Vec<U>,
-    }
-
-    let mut ice = Vec::new();
-    match serde_json::from_slice::<C>(&config) {
-        Ok(mut c) => {
-            for mut u in c.ice_servers.drain(..) {
-                ice.append(&mut u.urls);
-            }
-        }
-        Err(err) => tracing::error!(?err, "failed to parse iceServers"),
-    }
-
     let init_config = tx5_core::Tx5InitConfig::get();
 
     // TODO - max_message_size?
-    let config = datachannel::RtcConfig::new::<String>(&ice)
-        .port_range_begin(init_config.ephemeral_udp_port_min)
-        .port_range_end(init_config.ephemeral_udp_port_max);
+    let config = datachannel::RtcConfig::new::<String>(
+        &config
+            .ice_servers
+            .iter()
+            .flat_map(|s| s.urls.iter())
+            .cloned()
+            .collect::<Vec<_>>(),
+    )
```

Comment:
https://github.com/paullouisageneau/libdatachannel/blob/456c2151b7a1568b3805397a2d016e905c18a42e/src/configuration.cpp#L43-L87

Ugh... this is going to be frustrating to test. The rust bindings seem to only expose [the "url" constructor](https://github.com/paullouisageneau/libdatachannel/blob/456c2151b7a1568b3805397a2d016e905c18a42e/include/rtc/configuration.hpp#L23) for this config type.

So I guess we'll need to translate:

```json
iceServers: [{
  "urls": ["stun:one", "stun:two"],
  "credential": "abc",
  "username": "def",
}]
```

Into:

```json
iceServers: [
  "stun://abc:def@one",
  "stun://abc:def@two",
]
```

Reply:
This could be a follow-on at a later time.

Reply:
Yeah I think given that we weren't doing anything with credentials here before it doesn't need to happen on this PR. I think this can probably wait until somebody asks for it though because at least for now, we don't need TURN.